packages/jni-swig-stub/realm.i

```diff
@@ -179,63 +210,68 @@ struct realm_size_t {
 // How to
 // - delete all listeners
 %typemap(directorin, descriptor="J") const void* "*(const void **)&$input = $1;"
-%feature("director") NotificationCallback;
-// Ensures that the C++ callback object keeps a strong reference to the Java equivalent
-SWIG_DIRECTOR_OWNED(NotificationCallback)
+
 %inline %{
-class NotificationCallback {
-public:
-    virtual void onChange(const void*) {}
-    virtual ~NotificationCallback() {}
-};
-realm_notification_token_t* realm_object_add_notification_callbackJNI(
-        realm_object_t* object,
-        NotificationCallback* callback
-        ) {
+realm_notification_token_t*  register_results_notification_cb(realm_results_t* results, jobject callback) {
+        using namespace realm::jni_util;
+    auto jenv = realm::jni_util::get_env();
+    static jclass notification_class = jenv->FindClass("io/realm/interop/NotificationCallback");
+    static jmethodID on_change_method = jenv->GetMethodID(notification_class, "onChange", "(J)V");
+
+        return realm_results_add_notification_callback(
+                results,
+                // Use the callback as user data
+                static_cast<jobject>(get_env()->NewGlobalRef(callback)),
+        [](void *userdata) {
+            get_env(true)->DeleteGlobalRef(static_cast<jobject>(userdata));
+        },
+        // change callback
+        [](void* userdata, const realm_collection_changes_t* changes) {
+            auto jenv = realm::jni_util::get_env(true);
+            if (jenv->ExceptionCheck()) {
```

Comment:
What is the intention with this? Should it have been after invoking the callback? Or is the intention to identify issues with resolving the jenv??

Reply:
In general, we should check for pending exceptions before making another JNI call (in our case invoking the callback)